src/renderer/base/FontCache.h

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <dwrite_3.h>
+
+namespace Microsoft::Console::Render::FontCache
+{
+    namespace details
+    {
+        struct Cache
+        {
+            void Invalidate() noexcept
+            {
+                _lock.lock();
+                _fontCollection.reset();
+                _lock.unlock();
+            }
+
+            wil::com_ptr<IDWriteFontCollection> Get(bool forceUpdate)
+            {
+                std::lock_guard guard{ _lock };
+
+                if (!_fontCollection || forceUpdate)
+                {
+                    _init(forceUpdate);
+                    FAIL_FAST_IF(!_fontCollection);
```

Comment:
The PR that introduced all these fail-fasts into our code base made a pretty good point which I've never considered before: Fail-fasts make invariant-violations very obvious and easy to debug in our code base.
While exceptions are supposed to be just as easy to use, the existence of COM makes that a bit more difficult, because of their catch-all clauses.

So this fail-fast here is because I want to `assert()` that `_init()` really did create a `_fontCollection`, even if someone where to modify the code in the future. It ensures we don't return a null-pointer basically.

Alternatively I'd use `assert()` here or throw an exception.

Reply:
Sounds great to me -- I was imagining the failfasts that we used to have in buffer motion/coordinate walk that blew up the product for folks anyway.